From db9e77bfa046aeaaba711fdd066ba0eaf85293ac Mon Sep 17 00:00:00 2001
From: Joerg Bornemann <joerg.bornemann@qt.io>
Date: Mon, 29 Sep 2025 15:40:05 +0200
Subject: [PATCH 298/553] Doc: Extend example on QT_DEPLOY_QML_DIR
 documentation page

Make the example a QtQuick app that uses QT_DEPLOY_QML_DIR implicitly.
Remove the modification of QT_DEPLOY_QML_DIR from the non-QtQuick
example.

Fixes: QTBUG-121879
Change-Id: I4bf7afe85273cf1162d8bb27277b16cc9f664e6c
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit ecb44f06b0c6ce15126e90539a605fcb2a35feba)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../doc/src/cmake/cmake-deploy-variables.qdoc |  2 +-
 .../cmake-deploy-modified-qml-dir.qdocinc     | 31 +++++++++++++++++++
 ...ke-deploy-modified-variable-values.qdocinc |  1 -
 3 files changed, 32 insertions(+), 2 deletions(-)
 create mode 100644 src/corelib/doc/src/includes/cmake-deploy-modified-qml-dir.qdocinc

diff --git a/src/corelib/doc/src/cmake/cmake-deploy-variables.qdoc b/src/corelib/doc/src/cmake/cmake-deploy-variables.qdoc
index ac5094e7cb2..39027917e93 100644
--- a/src/corelib/doc/src/cmake/cmake-deploy-variables.qdoc
+++ b/src/corelib/doc/src/cmake/cmake-deploy-variables.qdoc
@@ -239,7 +239,7 @@ to be deployed to different locations within the app bundle.
 
 \section1 Example
 
-\include cmake-deploy-modified-variable-values.qdocinc
+\include cmake-deploy-modified-qml-dir.qdocinc
 
 \sa QT_DEPLOY_SUPPORT, QT_DEPLOY_PREFIX, QT_DEPLOY_BIN_DIR,
     QT_DEPLOY_LIBEXEC_DIR, QT_DEPLOY_LIB_DIR, QT_DEPLOY_PLUGINS_DIR,
diff --git a/src/corelib/doc/src/includes/cmake-deploy-modified-qml-dir.qdocinc b/src/corelib/doc/src/includes/cmake-deploy-modified-qml-dir.qdocinc
new file mode 100644
index 00000000000..4224426317a
--- /dev/null
+++ b/src/corelib/doc/src/includes/cmake-deploy-modified-qml-dir.qdocinc
@@ -0,0 +1,31 @@
+\badcode
+cmake_minimum_required(VERSION 3.16...3.22)
+project(MyThings)
+
+find_package(Qt6 REQUIRED COMPONENTS Quick)
+qt_standard_project_setup()
+
+qt_add_executable(MyApp main.cpp)
+qt_add_qml_module(MyApp
+    URI myapp
+    QML_FILES main.qml
+)
+
+set(deploy_script "${CMAKE_CURRENT_BINARY_DIR}/deploy_MyApp.cmake")
+file(GENERATE OUTPUT ${deploy_script} CONTENT "
+
+set(QT_DEPLOY_QML_DIR \"myqmldir\")
+
+include(\"${QT_DEPLOY_SUPPORT}\")
+
+qt_deploy_qml_imports(
+    TARGET MyApp
+    PLUGINS_FOUND plugins_found
+)
+qt_deploy_runtime_dependencies(
+    EXECUTABLE \"\${QT_DEPLOY_BIN_DIR}/$<TARGET_FILE_NAME:MyApp>\"
+    ADDITIONAL_MODULES \${plugins_found}
+)")
+
+install(SCRIPT ${deploy_script})
+\endcode
diff --git a/src/corelib/doc/src/includes/cmake-deploy-modified-variable-values.qdocinc b/src/corelib/doc/src/includes/cmake-deploy-modified-variable-values.qdocinc
index 632ddf756f7..45555c481b5 100644
--- a/src/corelib/doc/src/includes/cmake-deploy-modified-variable-values.qdocinc
+++ b/src/corelib/doc/src/includes/cmake-deploy-modified-variable-values.qdocinc
@@ -17,7 +17,6 @@ set(deploy_script "${CMAKE_CURRENT_BINARY_DIR}/deploy_MyApp.cmake")
 file(GENERATE OUTPUT ${deploy_script} CONTENT "
 
 set(QT_DEPLOY_PLUGINS_DIR \"mypluginsdir\")
-set(QT_DEPLOY_QML_DIR \"myqmldir\")
 set(QT_DEPLOY_TRANSLATIONS_DIR \"i18n\")
 
 include(\"${QT_DEPLOY_SUPPORT}\")
-- 
2.51.2

